source/MapPlanetCard.cpp

```diff
@@ -0,0 +1,247 @@
+/* MapPlanet.cpp
+Copyright (c) 2022 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "MapPlanetCard.h"
+
+#include "Color.h"
+#include "FillShader.h"
+#include "text/Font.h"
+#include "text/FontSet.h"
+#include "GameData.h"
+#include "Government.h"
+#include "Point.h"
+#include "PointerShader.h"
+#include "Politics.h"
+#include "Planet.h"
+#include "Interface.h"
+#include "Screen.h"
+#include "SpriteShader.h"
+#include "StellarObject.h"
+#include "text/WrappedText.h"
+
+using namespace std;
+
+vector<MapPlanetCard *> MapPlanetCard::cards{};
+double MapPlanetCard::scroll = 0.;
+
+
+
+MapPlanetCard::MapPlanetCard(const StellarObject &object, bool hasVisited) : hasVisited(hasVisited), planetName(object.Name())
+{
+	number = cards.size();
+	cards.emplace_back(this);
+	const Planet *planet = object.GetPlanet();
+	hasSpaceport = planet->HasSpaceport();
+	hasShipyard = planet->HasShipyard();
+	hasOutfitter = planet->HasOutfitter();
+
+	reputationLabel = !hasSpaceport ? "No Spaceport" :
+		GameData::GetPolitics().HasDominated(planet) ? "Dominated" :
+		planet->GetGovernment()->IsEnemy() ? "Hostile" :
+		planet->CanLand() ? "Friendly" : "Restricted";
+	
+	sprite = object.GetSprite();
+
+	const Interface* planetCardInterface = GameData::Interfaces().Get("map planet card");
+	float planetIconMaxSize = static_cast<float>(planetCardInterface->GetValue("planet icon max size"));
+	spriteScale = min(.5f, min((planetIconMaxSize) / sprite->Width(), (planetIconMaxSize) / sprite->Height()));
+}
+
+
+
+MapPlanetCard::~MapPlanetCard()
```

Comment:
Should I keep this destructor given I dont expect one to remove only one element? (They are meant to be all removed as one)